sympy/solvers/tests/test_diophantine.py

```diff
@@ -583,6 +582,58 @@ def check_solutions(eq):
 
     return okay
 
+
+def check_linear(eq, param=symbols("t", Integer=True)):
```

Comment:
This function seems to check only if the result satisfies the equation. It does not help to verify that it would be a full solution in the way the original tests did.